### ISSUE TITLE
MasterDetail - Updating a detail "clones" the record

### ISSUE THREAD

**Issue by danlem** [volkanceylan/Serenity#2917]
Hi,
I have a masterdetail relation in a form and I can insert or delete detail rows just fine.
But when I update a detail row and press the update button, it adds a copy of the edited record at the end of the "childrens" grid.

Any help would be appreciated.

Thanks

Original list in master dialog
![image](https://user-images.githubusercontent.com/16088184/32755470-749bbd3c-c8a3-11e7-9295-c397a5902491.png)

Edit detail #45 and update(ID with hidden attribute in detailcolumns.cs)
![image](https://user-images.githubusercontent.com/16088184/32755463-6971cbe0-c8a3-11e7-9b9f-105b0012591e.png)

Now I have 2 rows with same values
![image](https://user-images.githubusercontent.com/16088184/32755446-5a1842be-c8a3-11e7-9fb6-82d56560ead4.png)

Declaration in *MasterRow.cs

`
[MatierePremiereEnfantsFilEditor]
        [DisplayName("Wires"), MasterDetailRelation(foreignKey: "ID_MatierePremiere", IncludeColumns = "IQcWireExApprouveParDisplayName, IQcWireTeknaApprouveParDisplayName"), NotMapped]
        public List<MatierePremiereEnfantsRow> listeFils
        {
            get { return Fields.listeFils[this]; }
            set { Fields.listeFils[this] = value; }
        }
`

Declaration in *MasterForm.cs 

`
        [Category("Wires"), CssClass("cat-wires")]
        public List<Entities.MatierePremiereEnfantsRow> listeFils { get; set; }
`

Declaration in DetailRow.cs (generated by sergen)

```
// Detail ID
[DisplayName("Id Matiere Premiere Enfant"), Column("ID_MatierePremiereEnfant"), Identity]
        public Int32? IdMatierePremiereEnfant
        {
            get { return Fields.IdMatierePremiereEnfant[this]; }
            set { Fields.IdMatierePremiereEnfant[this] = value; }
        }

// Master ID
[DisplayName("Id Matiere Premiere"), Column("ID_MatierePremiere"), NotNull, ForeignKey("[dbo].[MatieresPremieres]", "ID_MatierePremiere"), LeftJoin("jIdMatierePremiere"), TextualField("IdMatierePremiereVcIdentLot")]
        public Int32? IdMatierePremiere
        {
            get { return Fields.IdMatierePremiere[this]; }
            set { Fields.IdMatierePremiere[this] = value; }
        }
```

**Comment by alexdaywid** [comment]
Hi friend, I believe this can help you.

var tipoDesconto: number = parseInt(this.form.TipoDesconto.value);
  var desconto: number = this.form.Desconto.value;

                       parcelas.push({                          
                            TipoDesconto: tipoDesconto,
                            Desconto: desconto,
                        });

**Comment by danlem** [comment]
Getting closer.

The problem seems to come from the GridEditorBase.ts file in the save function.

I added the following lines and got these outputs

`
console.log(row);
{TiIdentNoRouleau: 45, FPoidsEtiquette: null, FPoidsReel: null, FQcWireExAl: null, FQcWireExV: null, …}
DtCreation:"2017-11-13T16:50:00.000"
IdMatierePremiereEnfant:26
TiIdentNoRouleau:45
VcNotes:""
__id:"`2"

console.log(this.getIdProperty());
__id

console.log(this.id(row));
undefined
`

After that I tried console.log(Object.keys(row)); but the __id key is not present in the output.
I did the same thing in the northwind demo and the __id property was present in the output.

Any idea on why the __id key is not "detected" in the keys but is not enumerable ?

Thanks
